### PR TITLE
Update instructions for integration with Carthage

### PR DESCRIPTION
**Pull request by andrei-gorin**
Added more precise instructions for integration with Carthage
Updated dependency definition with explicit tag

IOS-250

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,66 @@
 [![Build Status](https://app.bitrise.io/app/fae70b8f60741639/status.svg?token=kHepFoJ1pOZ2YZOv6XIlFw)](https://app.bitrise.io/app/fae70b8f60741639)
 
-## Quick Start
 
-1. Get Carthage by running `brew install carthage`
-2. Run `carthage update`
+# Integration of SalemoveSDK using Carthage
+
+#### Demoes minimal setup required to get the SalemoveSDK/GliaSDK integrated using Carthage pacakge manager.
```
Comment (ostap0207) [inline README.md]: Something wrong with the number of `#`. `####` should be only used in the after `###` and so on.
If you want to highlight it you can make it **bold**.

File: README.md [hunk 2]
```diff
@@ -1,6 +1,66 @@
...
+
+### Sample was created with 
+* XCode 10.2.1 (`/usr/bin/xcodebuild -version`)
+* Swift 5.0.1 (`carthage version`)
+* Carthage 0.33.0 (`xcrun swift -version`)
+
+### Integration steps
+
+#### 1) Ensure directions in Cartfile are pulling explicitly from tags
+* That way you will always get the Released state of the SDK
```
Comment (ostap0207) [inline README.md]: We try to follow https://developers.google.com/style/tense and avoid using `will` as it can be replaced with present tense most of the time:

`This way you always get the Released state of the SDK`.

File: README.md [hunk 3]
```diff
@@ -1,6 +1,66 @@
...
+* Example: `github "Salemove/ios-bundle" "0.16.0"`
+
+#### 2) Get and build dependencies
+* Run `carthage update --no-use-binaries --platform iOS`
+	* Build all dependencies locally to ensure compatibility with your environment
+	* This will take some time to finsih.
```
Comment (ostap0207) [inline README.md]: This takes some time to finish.

File: README.md [hunk 4]
```diff
@@ -1,6 +1,66 @@
...
+* Known issues:
+	* 1) `Skipped building ios-webrtc due to the error`
+	* 2) `Skipped building ios-bundle due to the error`
+	* Both errors show up because of prebuilt frameworks for *ios-webrtc* and *ios-bundle*. These framework already exist and do not need to be built. And as they already exist (in `Carthage/checkouts` directory) then we do not need to wory about these errors.
+* NOTE: If command still fails due some version mismatch or incompatibility issues, then try delete the `Cartfile.resolved` file and `Carthage` directory and run the command again.
```
Comment (ostap0207) [inline README.md]: If **the** command still fails due **to** some

Comment (ostap0207) [inline README.md]: the try **to** delete

File: README.md [hunk 5]
```diff
@@ -1,6 +1,66 @@
...
+
+#### 3) Link built libraries
+* Carthage update command will pull and build dependencies into *Carthage* folder
```
Comment (ostap0207) [inline README.md]: Could you split the first 3 points into just description and leave the last two. The reason is that currently the instructions of what integrator should do is mixed with what Carthage does.

```
Carthage update command will pull and build dependencies into *Carthage* folder. Pulled files are in *Carthage/checkouts* folder and *Carthage/build* files are in build folder. There you will find different `*.framework` files which you need to link into your xcode project.

To link the files:
* In Xcode, click your project in the *Project Navigator*. Select your *target*, choose the *General* tab at the top, and scroll down to the *Linked Frameworks and Libraries* section at the bottom.
* Drag and drop all your `*.framework` files (except `RxTest.framework` which is not needed in Apps directly) from Carthage/checkouts* and *Carthage/build* folders to that section in xcode.
```

Also, try to replace `will` with the present tense.

**Comment by andrei-gorin** [comment]
I will run the spellchecker.

File: README.md [hunk 6]
```diff
@@ -1,6 +1,66 @@
...
+* Pulled files are in *Carthage/checkouts* folder and *Carthage/build* files are in build folder
+* There you will find different `*.framework` files which you need to link into your xcode project.
+* In Xcode, click your project in the *Project Navigator*. Select your *target*, choose the *General* tab at the top, and scroll down to the *Linked Frameworks and Libraries* section at the bottom.
+* Drag and drop all your `*.framework` files (except `RxTest.framework` which is not needed in Apps directly) from *Carthage/checkouts* and *Carthage/build* folders to that section in xcode.
```
Comment (gugalo) [inline README.md]: > (except `RxTest.framework` which is not needed in Apps directly)

I think just ''(except `RxTest.framework`)" is better.

File: README.md (hunk 5 above)
Comment (andrei-gorin) [reply README.md]: Do you mean completely remove the last two bullets or just split to separate paragraphs?
Also, basically it is all Carthage. So if we want to remove all Carthage know-how from README then I guess it is enough to update the `carthage update` command to `carthage update --no-use-binaries --platform iOS`
But on the other hand, it looks better to see complete instructions.

Comment (ostap0207) [reply README.md]: I mean to split them as I showed above as the first 3 bullet points are not instructions, but just describe the current state. The last 2 bullet points are actually the instructions and can stay as-is:

```
Carthage update command will pull and build dependencies into *Carthage* folder. Pulled files are in *Carthage/checkouts* folder and *Carthage/build* files are in build folder. There you will find different `*.framework` files which you need to link into your xcode project.

To link the files:
* In Xcode, click your project in the *Project Navigator*. Select your *target*, choose the *General* tab at the top, and scroll down to the *Linked Frameworks and Libraries* section at the bottom.
* Drag and drop all your
```

File: README.md [hunk 7]
```diff
@@ -1,6 +1,68 @@
...
+	* Builds all dependencies locally to ensure compatibility with your environment
+	* This takes some time to finish.
+* Known issues:
+	* 1) `Skipped building ios-webrtc due to the error`
+	* 2) `Skipped building ios-bundle due to the error`
+	* Both errors show up because of prebuilt frameworks for *ios-webrtc* and *ios-bundle*. These frameworks already exist and do not need to be built. And as they already exist (in `Carthage/checkouts` directory) then we do not need to worry about these errors.
+* NOTE: If the command still fails due to some version mismatch or incompatibility issues, then try deleting the `Cartfile.resolved` file and `Carthage` directory and run the command again.
+
+#### 3) Link built libraries
+* Carthage update command pulls and builds dependencies into *Carthage* folder
```
Comment (ostap0207) [inline README.md]: If it's just a description then it does not have to be a bullet point, can be just a sentence.

File: README.md [hunk 8]
```diff
@@ -1,6 +1,68 @@
...
+
+#### 1) Ensure directions in Cartfile are pulling explicitly from tags
+* This way you always get the Released state of the SDK
+* Example: `github "Salemove/ios-bundle" "0.16.0"`
+
+#### 2) Get and build dependencies
+* Run `carthage update --no-use-binaries --platform iOS`
+	* Builds all dependencies locally to ensure compatibility with your environment
+	* This takes some time to finish.
+* Known issues:
```
Comment (ostap0207) [inline README.md]: I think bullet points are a bit overused where simple paragraphs are enough.

I think bullet points should be used for actual lists of related things (like commands to run) and other things like `Known issues:` should be just a separate paragraph same as `NOTE:`.

Comment (andrei-gorin) [reply README.md]: I do not understand this constraint. Bullets are bullets. Indented Bullets are "sub topics" to parent Bullet. And that's mostly it.

File: README.md (hunk 7 above)
Comment (andrei-gorin) [reply README.md]: Bulleting, in general, is used to make larger parts of text easier to read. Without bullets, it would be just another textbook.

File: README.md [hunk 9]
```diff
@@ -1,6 +1,68 @@
...
+
+
+### Sample was created with 
+* XCode 10.2.1 (`/usr/bin/xcodebuild -version`)
+* Swift 5.0.1 (`carthage version`)
+* Carthage 0.33.0 (`xcrun swift -version`)
+
+### Integration steps
+
+#### 1) Ensure directions in Cartfile are pulling explicitly from tags
```
Comment (gugalo) [inline README.md]: This line was a bit confusing for me (but I am new to Carthage). If it is possible to rephrase it to make it more clear it would be nice. Maybe something like this if it makes sense:
> Ensure that Salemove SDK dependency in your Cartfile specifies version tag

File: README.md (hunk 7 above)
Comment (ostap0207) [reply README.md]: What I mean is that for top-level things you don't need a bullet point. You can still paragraphs to do exactly the same job.

If you put bullet points everywhere then it's essentially bullet point no-nowhere and defeats the purpose of having them.

> Without bullets, it would be just another textbook

Not saying to remove them, but use them only when you actually have a list like in p.4 it's OK.

File: README.md (hunk 8 above)
Comment (ostap0207) [reply README.md]: They are not exactly:

> In typography, a bullet ( • ) is a typographical symbol or glyph used to introduce items in a list.

I am OK with using bullet points, but again, they are overused when normal paragraphs would be enough.
For example, you can rewrite the following as while having the same if not better readibility:

```
Run `carthage update --no-use-binaries --platform iOS:
  * Builds all dependencies locally to ensure compatibility with your environment
  * This takes some time to finish.

Known issues:
  * 1) `Skipped building ios-webrtc due to the error`
  * 2) `Skipped building ios-bundle due to the error`
  * Both errors show up because of prebuilt frameworks

NOTE: If the command still fails due to some version mismatch or incompatibility issues, then try deleting the `Cartfile.resolved` file and `Carthage` directory and run the command again.

```

If this is still not clear then let's talk in slack when I am back from vacation.
